pkg/store/event/event.go

```diff
@@ -262,44 +262,83 @@ func (e *eventStore) DeleteIncompleteEvents(_ context.Context) error {
 		Delete(core.Event{}).Error
 }
 
-// ListByFilter returns an event list by the podName, podNamespace, experimentName, experimentNamespace, uid and the startTime.
-func (e *eventStore) ListByFilter(_ context.Context, podName string, podNamespace string, experimentName string, experimentNamespace string, uid string, startTimeStr string) ([]*core.Event, error) {
+// ListByFilter returns an event list by podName, podNamespace, experimentName, experimentNamespace, uid, kind, startTime and finishTime.
+func (e *eventStore) ListByFilter(_ context.Context, filter core.Filter) ([]*core.Event, error) {
 	var resList []*core.Event
 	var err error
-	var startTime time.Time
-
-	if podName != "" {
-		resList, err = e.ListByPod(context.Background(), podNamespace, podName)
-	} else if podNamespace != "" {
-		resList, err = e.ListByNamespace(context.Background(), podNamespace)
-	} else if uid != "" {
-		resList, err = e.ListByUID(context.Background(), uid)
+	var startTimeBegin, startTimeEnd, finishTimeBegin, finishTimeEnd time.Time
+
+	if filter.PodName != "" {
+		resList, err = e.ListByPod(context.Background(), filter.PodNamespace, filter.PodName)
+	} else if filter.PodNamespace != "" {
+		resList, err = e.ListByNamespace(context.Background(), filter.PodNamespace)
 	} else {
-		resList, err = e.List(context.Background())
+		query, args := constructQueryArgs(filter.ExperimentName, filter.ExperimentNamespace, filter.UID, filter.Kind)
+		// List all events
+		if len(args) == 0 {
+			if err := e.db.Model(core.Event{}).Find(&resList).Error; err != nil && !gorm.IsRecordNotFoundError(err) {
+				return resList, err
+			}
+		} else {
+			if err := e.db.Where(query, args...).Find(&resList).Error; err != nil &&
+				!gorm.IsRecordNotFoundError(err) {
+				return resList, err
+			}
+		}
 	}
 	if err != nil {
 		return resList, err
 	}
 
-	if startTimeStr != "" {
-		startTime, err = time.Parse(time.RFC3339, strings.Replace(startTimeStr, " ", "+", -1))
+	if filter.StartTimeBeginStr != "" {
```

Comment:
If we need to check these variables, we should check them before selecting data from db.